### ISSUE TITLE
Local LLM support please

### ISSUE THREAD

**Issue by razvanab** [lamm-mit/PDF2Audio#1]
Can you please add local LLM support, please?
Ollama support will be nice too.
Thank you.

**Comment by si0103** [comment]
Adding to the above: it wold be great to allow for referencing folders that contain models already.
Thanks a bunch!

**Comment by KnutJaegersberg** [comment]
yeah it's not really open source if it doesn't allow local inference with some of the recent open models

**Comment by dareima** [comment]
> yeah it's not really open source if it doesn't allow local inference with some of the recent open models

I think it would be unfair to say it like that. It is open source. You could add support for local models yourself.
It would certainly benefit this great project to have it by design though, which might very well be the case soon.

**Comment by dareima** [comment]
> Can you please add local LLM support, please? Ollama support will be nice too. Thank you.

@razvanab did you check this?

![image](https://github.com/user-attachments/assets/50363727-59e4-4eee-9aca-50d24d3d5522)

**Comment by lombarthus** [comment]
> > Can you please add local LLM support, please? Ollama support will be nice too. Thank you.
> 
> @razvanab did you check this?
> 
> ![image](https://private-user-images.githubusercontent.com/64737959/370325257-50363727-59e4-4eee-9aca-50d24d3d5522.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyMTMyNzEsIm5iZiI6MTcyNzIxMjk3MSwicGF0aCI6Ii82NDczNzk1OS8zNzAzMjUyNTctNTAzNjM3MjctNTllNC00ZWVlLTlhY2EtNTBkMjRkM2Q1NTIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI0VDIxMjI1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFhMzkxN2M3YmU3ZTdjM2Y4MDA1OTg4ODdiNjUwYzc1MjU3YzhjN2E0YjBjOTFjMDVkN2Y4NzE1NjE0MjViOWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2zaWCkLJImbyMkWvopxsac5rvnNECLsRhDpNVkV8vV4)


Yes, but then it still seems to require an OpenAI API Key.

**Comment by rushizirpe** [comment]
Try:
`export OPENAI_API_KEY="DUMMY_KEY"`
And then:
`python app.py`

**Comment by razvanab** [comment]
Yes, I did check that, and I didn't have much success.

**Comment by rushizirpe** [comment]
Here's how I am able to resolve:

`pip install llama-cpp-python[server]`
`wget https://huggingface.co/TheBloke/zephyr-7B-beta-GGUF/resolve/main/zephyr-7b-beta.Q6_K.gguf`
`python3 -m llama_cpp.server --model ./zephyr-7b-beta.Q6_K.gguf --host 127.0.0.1 --port 8888`
`export OPENAI_API_KEY=DUMMY`
`python app.py`

Custom Base API: `http://localhost:8888/v1`
Text Generation Model: `openai/custom_model`

**Comment by jsCarvalhoNeto** [comment]
My dear, could you detail these procedures for laymen?

**Comment by mfoldes** [comment]
> My dear, could you detail these procedures for laymen?

If you're going to run a local model, you'll have to download it and run it locally. This person is giving you commands to run to do that. 

Here are his instructions in a shell script

```bash
#!/bin/bash

# Install the required Python package
pip install llama-cpp-python[server]

# Download the model file
wget https://huggingface.co/TheBloke/zephyr-7B-beta-GGUF/resolve/main/zephyr-7b-beta.Q6_K.gguf

# Start the server with the specified model
python3 -m llama_cpp.server --model ./zephyr-7b-beta.Q6_K.gguf --host 127.0.0.1 --port 8888 &

# Export the API key
export OPENAI_API_KEY=DUMMY

# Run the application
python app.py
```

**Comment by jsCarvalhoNeto** [comment]
In the third step when running "python3 -m llama_cpp.server --model ./zephyr-7b-beta.Q6_K.gguf --host 127.0.0.1 --port 8888" I get the following message: C:\Users\santo\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\python.exe: Error while finding module specification for 'llama_cpp.server' (ModuleNotFoundError: No module named 'llama_cpp')" Could someone help?

**Comment by moebiussurfing** [comment]
try `python -m llama_cpp.server --model ./zephyr-7b-beta.Q6_K.gguf --host 127.0.0.1 --port 8888 --n_ctx 16384`
notice the last argument if you need more context size.

**Comment by moebiussurfing** [comment]
I followed the steps but I got an API KEY error (even using the DUMMY alias):
``warning Error code: 401 - {'error': {'message': 'Incorrect API key provided: DUMMY. You can find your API key at https://platform.openai.com/account/api-keys.', 'type': 'invalid_request_error', 'param': None, 'code': 'invalid_api_key'}}``
